src/frontend/src/catalog/schema_catalog.rs

```diff
@@ -64,22 +64,38 @@ impl SchemaCatalog {
         self.source_by_name.remove(&name).unwrap();
     }
 
-    // Use associated source to filter table.
-    pub fn get_all_table_names(&self) -> Vec<String> {
+    pub fn iter_table(&self) -> impl Iterator<Item = &TableCatalog> {
         self.table_by_name
             .iter()
             .filter(|(_, v)| v.associated_source_id.is_some())
-            .map(|(k, _)| k.clone())
-            .collect_vec()
+            .map(|(_, v)| v)
     }
 
-    // Use associated source to filter mv.
-    pub fn get_all_mv_names(&self) -> Vec<String> {
+    pub fn iter_mv(&self) -> impl Iterator<Item = &TableCatalog> {
         self.table_by_name
             .iter()
             .filter(|(_, v)| v.associated_source_id.is_none())
-            .map(|(k, _)| k.clone())
-            .collect_vec()
+            .map(|(_, v)| v)
+    }
+
+    pub fn iter_source(&self) -> impl Iterator<Item = &SourceCatalog> {
+        self.source_by_name
+            .iter()
+            .filter(|(name, v)| {
+                matches!(v.source_type, SourceType::Source)
+                    && self.table_by_name.get(*name).is_none()
+            })
+            .map(|(_, v)| v)
+    }
+
+    pub fn iter_materialized_source(&self) -> impl Iterator<Item = &SourceCatalog> {
+        self.source_by_name
+            .iter()
+            .filter(|(name, v)| {
+                matches!(v.source_type, SourceType::Source)
```

Comment:
```suggestion
                matches!(v.source_type, SourceType::Table)
```
Currently our materialized source only contains table source, maybe it should be `SourceType::Table`?

Reply:
When we create a table, it will be identified as `SourceType::Table`, but if it's created as a MATERIALIZED SOURCE, we use `SourceType::Source` to identify that it couldn't accept an INSERT statement.

Reply:
`Create Table` => (`SourceType::Table`, materialized)
`Create Materialized Source` => (`SourceType::Source`, materialized)
`Create Source` => (`SourceType::Source`, not materialized)

If so, `Show Source` and `Show Materialized Source` will return same result. Seems like we need a new field `is_materialized` to indicate whether the source is materialized? 

Besides, the source in type `SourceType::Table` is naturally materialized. If we treat `Create Table` as an alias of something like `Create Materialized Source table_t`, should it be return when calling `Create Materialized Source`? Or just showing it as a result of `Show Tables`(prefer this for not confusing our customer)?